core/src/main/scala/com/nvidia/spark/rapids/tool/tuning/AutoTuner.scala

```diff
@@ -356,6 +356,9 @@ abstract class AutoTuner(
   protected val limitedLogicRecommendations: mutable.HashSet[String] = mutable.HashSet[String]()
   // When enabled, the profiler recommendations should only include updated settings.
   private var filterByUpdatedPropertiesEnabled: Boolean = true
+  // OS reserved memory for system processes, configurable via tuning configs
+  private lazy val osReservedMemory = tuningConfigs.getEntry("OS_RESERVED_MEM")
```

Comment:
There exists a platform specific property `Platform.fractionOfSystemMemoryForExecutors`, which is used to reduce the available memory for executors in AutoTuner (see initialization of `actualMemForExec`)

Should we use something like:

total available memory for executors:
```
totalMemForExecutors = totalMemForExecExpr.apply()

if osReservedMemory > 0m:
    totalMemForExecutors - osReservedMemory
else:
    totalMemForExecutors * platform.fractionOfSystemMemoryForExecutors
```

Reply:
good catch, updated.